ui/tests/components/StatusCardList.test.tsx

```diff
@@ -0,0 +1,49 @@
+import { render, screen } from "@testing-library/react"
+import StatusCardList from "../../src/components/StatusCardList"
+import { StackStatus } from "../../src/hooks/useFetchStatuses"
+
+describe("StatusCardList", () => {
+  const statuses: StackStatus[] = [
+    { Name: "Foobar", Error: "", Revision: "1.0.0", RepoURL: "https://www.url1.com" },
+    { Name: "FooFoo", Error: "", Revision: "2.0.0", RepoURL: "https://www.url2.com" },
+    { Name: "Boobaz", Error: "Oh no!!!", Revision: "2.0.0", RepoURL: "https://www.url3.com" }
+  ]
+
+  it("should render no statuses if the list of statuses is empty", () => {
+    render(<StatusCardList statuses={[]} query="" />)
+    expect(screen.getByText(/No items/i)).toBeInTheDocument()
+  })
+
+  it("should render a list of statuses", () => {
+    render(<StatusCardList statuses={statuses} query="" />)
+    for (const status of statuses) {
+      expect(screen.getByText(status.Name)).toBeInTheDocument()
+    }
+  })
+
+  it("should filter out the whole list of statuses if query is not found", () => {
+    render(<StatusCardList statuses={statuses} query="NOT FOUND!!!" />)
+    expect(screen.queryByText("Foobar")).not.toBeInTheDocument()
+    expect(screen.queryByText("FooFoo")).not.toBeInTheDocument()
+    expect(screen.queryByText("Boobaz")).not.toBeInTheDocument()
```

Comment:
Could be a loop for brevity and consistency

Reply:
Done